imager/imager_images/templatetags/photo_extras.py

```diff
@@ -0,0 +1,20 @@
+from django import template
+from imager_images.models import Photo
+from django.db.models import Q
+
+
+register = template.Library()
+
+# Custom filter, return photos that are public, shared, or belong to a logged in user
+@register.filter
+def viewable(self, user):
+    return Photo.objects.filter(
+        Q(user=user) |
+        Q(published=Photo.SHARED) |
+        Q(published=Photo.PUBLIC)
+        )
+
+# Return photos belonging only to one user.
+@register.filter
+def viewable_user(self, user):
+    return Photo.objects.filter(Q(user=user))
```

Comment:
interesting.  Using template filters to get a set of photos is an approach I would not have thought of.  It's probably not the most efficient, but points for finding the path here!